lib/services/unified_resource.go

```diff
@@ -305,14 +307,18 @@ func (c *UnifiedResourceCache) GetUnifiedResourcesByIDs(ctx context.Context, ids
 				continue
 			}
 			resource := cache.resources[res.Value]
-			if matched := matchFn(resource); matched {
+			match, err := matchFn(resource)
+			if err != nil {
+				return trace.Wrap(err)
```

Comment:
Do we have any existing tests that cover this new behavior where a match fn can fail?